### PR TITLE
📖  Add documentation for `bento-social-share` and `BentoSocialShare`

### PR DESCRIPTION
**Pull request by caroqliu**
This PR:
- creates README.md for 1.0 version pertaining to web component (non-AMP) and react version usage
- renames the `SocialShare` preact component as `BentoSocialShare`

### CODE REVIEW

File: extensions/amp-social-share/1.0/README.md [hunk 1]
```diff
@@ -0,0 +1,597 @@
+---
+$category@: ads-analytics
+formats:
+  - websites
```
Comment (kvchari) [inline extensions/amp-social-share/1.0/README.md:4]: Do we need teaser text here?

**Review by kvchari — APPROVED** [review]

File: extensions/amp-social-share/1.0/README.md (hunk 1 above)
Comment (caroqliu) [reply extensions/amp-social-share/1.0/README.md:4]: Oh yeah, good call.

Comment (kvchari) [reply extensions/amp-social-share/1.0/README.md]: oh, i'd meant it was missing those attributes. should we be removing the frontmatter from the `<extension>/1.0/README.md` files?

Comment (caroqliu) [reply extensions/amp-social-share/1.0/README.md]: Yeah, I learned this morning that they aren't needed so I created #36121 to fix everything so far

**Pull request merged**